src/commands/project/pipeline.ts

```diff
@@ -66,6 +66,26 @@ export const checkDependencies = (projectPath: string): void => {
   }
 };
 
+export const validateRepoUrl= (
+  opts: CommandOptions,
+  gitOriginUrl: string
+): string => {
+  try {
+    let repoUrl = ''
+    if (!opts.repoUrl) {
+      if(!getRepositoryUrl(gitOriginUrl)) {
+        throw Error(`Repo url not defined. Are you in a spk project folder?`)
+      } else {
+        return repoUrl = getRepositoryUrl(gitOriginUrl)
+      }
+    } else {
+      return repoUrl = opts.repoUrl
+    }
+  } catch(err){
+    throw new Error(`Unable to obtain and validate the repo url.`)
+  }
+}
+
```

Comment:
just a stylistic comment - in some places there's uses of `;` at the end of every statement (see line 253 below), but there there isn't in this code block

I think we should _probably_ enable this eslint rule https://eslint.org/docs/rules/semi to settle it one way or another, but would just like to call it out as a difference

Reply:
Will enable eslint rule for consistency! Thanks!